app/controllers/keybase_proofs_controller.rb

```diff
@@ -0,0 +1,45 @@
+class KeybaseProofsController < ApplicationController
+  before_action :require_logged_in_user
+  before_action :check_new_params, only: :new
+  before_action :check_user_matches, only: :new
+
+  def new
+    @kb_username = params[:kb_username]
+    @kb_signature = params[:kb_signature]
+    @kb_ua = params[:kb_ua]
+    @kb_avatar = Keybase.avatar_url(@kb_username)
+  end
+
+  def create
+    kb_username = post_params[:kb_username]
+    kb_signature = post_params[:kb_signature]
+    kb_ua = post_params[:kb_ua]
+    if Keybase.proof_valid?(kb_username, kb_signature, @user.username)
+      @user.add_or_update_keybase_proof(kb_username, kb_signature)
+      @user.save!
+      return redirect_to Keybase.success_url(kb_username, kb_signature, kb_ua, @user.username)
+    else
+      flash[:error] = "Failed to connect your account to Keybase. Try again from Keybase."
+      return redirect_to settings_path
+    end
+  end
+
+private
+
+  def check_user_matches
+    unless @user.username.casecmp(params[:username]).zero?
```

Comment:
Would `casecmp?` match Keybase's desired behavior here? I like that it's a little clearer to use a boolean predicate than check for zero on a sorting operator, and it's nice that it supports Unicode for the non-english sister sites that have broadened the definition of `User::VALID_USERNAME`.

Reply:
`casecmp?` isn't until ruby 2.4.6 and we're on 2.3.3 :/
but i can extract this into a predicate so at least it's easier to read.